### PR TITLE
Add italian version

### PR DESCRIPTION
**Pull request by scara82**
This PR:

- Add Italian version
- Add english as default
- Add Italian transaltions
- Add link in the header to switch language
- Add locale params to all the links.

Trello Card [https://trello.com/c/y9Kd3Flu](url)

### CODE REVIEW

**Comment by apalski** [comment]
LGTM!

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,13 +1,13 @@
 require "application_responder"
 
 class ApplicationController < ActionController::Base
-  self.responder = ApplicationResponder
-  respond_to :html
-
   protect_from_forgery with: :exception
   before_action :set_locale
+  self.responder = ApplicationResponder
+  respond_to :html
```
Comment (elle) [inline app/controllers/application_controller.rb]: Why did you move the lines?

File: app/views/application/_language.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<% if I18n.locale == I18n.default_locale %>
+  <%= link_to "It", { :locale=>'it' } %>
```
Comment (elle) [inline app/views/application/_language.html.erb]: double quotes.
Also, can we use the `{ locale: "it" }` syntax?

File: app/views/application/_language.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+<% else %>
+  <%= link_to "En", { :locale=>'en' } %>
```
Comment (elle) [inline app/views/application/_language.html.erb]: Same ^

File: app/views/families/_family.html.erb [hunk 4]
```diff
@@ -8,6 +8,6 @@
   <td><%= family.bus_driver %></td>
   <td><%= family.animals %></td>
   <td><%= family.active %></td>
-  <td><%= link_to t("families.index.destroy"), family_path(family), method: :delete, class: "list-button" %></td>
-  <td><%= link_to t("families.index.edit"), edit_family_path(family), class: "list-button" %></td>
+  <td><%= link_to t("families.index.destroy"), family_path(family, locale: I18n.locale), method: :delete, class: "list-button" %></td>
```
Comment (elle) [inline app/views/families/_family.html.erb]: If you look at the curriculum, there is a way to add the locale to all the links by default.
In `ApplicationController`:

```
def default_url_options(options = {})
  { locale: I18n.locale }.merge(options)
end
```

File: config/routes.rb [hunk 5]
```diff
@@ -1,8 +1,10 @@
 Rails.application.routes.draw do
-  resources :families
-  resources :schools, only: [:index, :new, :create, :edit, :update, :destroy]
-  resources :trips, only: [:index, :new, :create, :edit, :update, :destroy]
-  resources :pages, only: [:show]
+  scope "(:locale)", locale: /en|it/ do
```
Comment (elle) [inline config/routes.rb:2]: What will happen if a user enters a locale that is not supported?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (scara82) [reply app/controllers/application_controller.rb]: I probably moved when the translation was not working. :) Moved back with fixup in sha `e728c8d`

File: app/views/application/_language.html.erb (hunk 2 above)
Comment (scara82) [reply app/views/application/_language.html.erb]: Done with fixup in sha `16c1673`

File: app/views/application/_language.html.erb (hunk 3 above)
Comment (scara82) [reply app/views/application/_language.html.erb]: Done with fixup in sha `16c1673`

File: app/views/families/_family.html.erb (hunk 4 above)
Comment (scara82) [reply app/views/families/_family.html.erb]: Done with fixup in sha `631fd59`

File: config/routes.rb (hunk 5 above)
Comment (scara82) [reply config/routes.rb:2]: @elle I thought that it would use the default one but now I checked...I couldn't find any solutions. I tried to modify the set_locale method but without success. I can set the locale to the default one but the path is still unknown...I'm sure there is an easy and straightforward way... 🙄

Comment (scara82) [reply config/routes.rb:2]: I will implement it in another PR. I will merge this for now. Thanks for the tip!

**Comment by scara82** [comment]
Thanks @apalski  and @elle for your suggestions. I created a Trello card to manage the case when another language rather than it/en is added in the url.

I'm merging now 🚀

File: app/views/application/_language.html.erb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<% if I18n.locale == I18n.default_locale %>
+  <%= link_to "It", { locale: "it" } %>
+<% else %>
+  <%= link_to "En", { locale: "en" } %>
+<%end%>
```
Comment (elle) [inline app/views/application/_language.html.erb:5]: How about displaying both languages? Thus removing the need to have an `if` block....
In that case, you will want to have an html class to indicate which language is currently selected...
